Specs/Scene/GltfFeatureMetadataLoaderSpec.js

```diff
@@ -0,0 +1,485 @@
+        when.resolve(image)
+      );
+
+      spyOn(Resource.prototype, "fetchJson").and.returnValue(
+        when.resolve(schemaJson)
+      );
+
+      var featureMetadataLoader = new GltfFeatureMetadataLoader({
+        gltf: gltfSchemaUri,
+        extension: extensionSchemaUri,
+        gltfResource: gltfResource,
+        baseResource: gltfResource,
+        supportedImageFormats: new SupportedImageFormats(),
+      });
+
+      featureMetadataLoader.load();
+
+      return pollToPromise(function () {
+        featureMetadataLoader.process(scene.frameState);
+        return featureMetadataLoader._state === ResourceLoaderState.READY;
+      }).then(function () {
+        return featureMetadataLoader.promise.then(function (
+          featureMetadataLoader
+        ) {
+          var featureMetadata = featureMetadataLoader.featureMetadata;
+          expect(Object.keys(featureMetadata.schema.classes).sort()).toEqual([
+            "building",
+            "map",
+            "ortho",
+            "tree",
+          ]);
+        });
+      });
+    });
+
+    it("destroys feature metadata", function () {
+      spyOn(Resource.prototype, "fetchArrayBuffer").and.returnValue(
+        when.resolve(buffer)
+      );
+
+      spyOn(Resource.prototype, "fetchImage").and.returnValue(
+        when.resolve(image)
+      );
+
+      spyOn(Resource.prototype, "fetchJson").and.returnValue(
+        when.resolve(schemaJson)
+      );
+
+      var destroyBufferView = spyOn(
+        GltfBufferViewLoader.prototype,
+        "destroy"
+      ).and.callThrough();
+
+      var destroyTexture = spyOn(
+        GltfTextureLoader.prototype,
+        "destroy"
+      ).and.callThrough();
+
+      var destroySchema = spyOn(
+        MetadataSchemaLoader.prototype,
+        "destroy"
+      ).and.callThrough();
+
+      var featureMetadataLoader = new GltfFeatureMetadataLoader({
+        gltf: gltf,
+        extension: extension,
+        gltfResource: gltfResource,
+        baseResource: gltfResource,
+        supportedImageFormats: new SupportedImageFormats(),
+      });
+
+      featureMetadataLoader.load();
+
+      return pollToPromise(function () {
+        featureMetadataLoader.process(scene.frameState);
+        return featureMetadataLoader._state === ResourceLoaderState.READY;
+      }).then(function () {
+        return featureMetadataLoader.promise.then(function (
+          featureMetadataLoader
+        ) {
+          expect(featureMetadataLoader.featureMetadata).toBeDefined();
+          expect(featureMetadataLoader.isDestroyed()).toBe(false);
+
+          featureMetadataLoader.destroy();
+
+          expect(featureMetadataLoader.featureMetadata).not.toBeDefined();
+          expect(featureMetadataLoader.isDestroyed()).toBe(true);
+
+          expect(destroyBufferView.calls.count()).toBe(6);
+          expect(destroyTexture.calls.count()).toBe(2);
+          expect(destroySchema.calls.count()).toBe(1);
+        });
+      });
+    });
+  },
```

Comment:
looking at the coverage, one thing you could test is that it handles the case where the object is destroyed before load finishes, like we did with the other loaders.